cobc/typeck.c

```diff
@@ -6354,21 +6369,29 @@ decimal_expand (cb_tree d, cb_tree x)
 		 * Set t, Y
 		 * OP d, t */
 		p = CB_BINARY_OP (x);
-		if ((p->op == 'c' || p->op == 'd')		/* Circular Shift */
-		 && CB_REF_OR_FIELD_P (p->x)) {
-			sz_shift = cb_int (CB_FIELD_PTR (p->x)->size);
-		} else {
-			sz_shift = cb_int1;
+
+		switch (p->op){
+		case '=': case '~': case '<': case '>': case '[': case ']':
+		case '!': case '&': case '|':
+			cb_error_x (x, _("condition expression found where decimal expression was expected"));
+			return cb_error_node;
+		case 'c':
+		case 'd':
```

Comment:
This is switching a `cb_binary_op_op` - it is likely more readable if not checking for the long names (and a lesser diff), I've just added a comment back and another in.